modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentShortcutResourceImpl.java

```diff
@@ -38,6 +49,19 @@ public DocumentShortcut getDocumentShortcut(Long documentShortcutId)
 			_dlAppService.getFileShortcut(documentShortcutId));
 	}
 
+	@Override
+	public Page<DocumentShortcut> getSiteDocumentShortcutsPage(Long siteId)
+		throws Exception {
+
+		List<FileShortcut> fileShortcuts = _dlAppService.getGroupFileShortcuts(
+			siteId);
+
+		return Page.of(
```

Comment:
@SamZiemer, is there no pagination? What if the user has hundreds/thousands of shortcuts?

Reply:
Thanks @jkappler, you make a good point, we definitely need pagination for instances of many many shortcuts, I've added that in some new commits.